courses/migrations/0031_populate_letter_grades.py

```diff
@@ -0,0 +1,32 @@
+# Generated by Django 3.2.18 on 2023-06-22 19:54
+import datetime
+
+import pytz
+from django.db import migrations
+
+from courses.utils import convert_to_letter
+
+
+def populate_letter_grade(apps, schema_editor):
+    """Populate the certificate_available_date from course run's end_date"""
+    CourseRunGrade = apps.get_model("courses", "CourseRunGrade")
+    grades = CourseRunGrade.objects.filter(
+        passed=True,
+        letter_grade__contains="Pass",
+        course_run__course__program__title__startswith="Data",
+        course_run__end_date__lt=datetime.datetime(2022, 9, 1, tzinfo=pytz.UTC),
+    )
+    for grade in grades:
+        grade.letter_grade = convert_to_letter(grade.grade)
+        grade.save_and_log()
```

Comment:
I am getting an error on `AttributeError: 'CourseRunGrade' object has no attribute 'save_and_log'`. Looks like this method isn't defined in a migration-defined model using `apps.get_mode`.